### PR TITLE
[ADMIN QOL] The Nuclear Device Code is now set automatically to a random value.

### PR DESCRIPTION
**Pull request by Nickvr628**
Whenever an admin wants to have the nuke be used, they have to manually edit the code to a random value. This is cumbersome, and since there is already a proc to randomize the code anyway, I just made it create one at roundstart. The code is a random 5 digit code like normal.

The odds of someone randomly guessing the code are essentially impossible, so that is not a concern.

All this really does is reduce a step for admins to let the captain/crew scuttle the station, because it is a hassle to manually varedit the self destruct device and change it to a 5 digit value and we might as well streamline it. This also lets admins without the power to edit variables approve self destruct requests, which removes cases of valid requests going unanswered because all the staff online are unable to set the code.

This does not break nuke ops, I checked, and it actually removes some back and forth shitcode by automatically giving the self destruct device a code regardless of nuke ops triggering or not.

 
:cl:  
tweak: Instead of admins needing to manually set it to use it, the station self destruct device now has a random passcode. Admins can quickly check and send the code to the station or ERTs.
/:cl:

### CODE REVIEW

**Comment by SomeguyManperson** [comment]
if you give something a value it WILL be used regardless of intent. It doesn't matter if it's a 5 digit fully random code if it's still something that really exists people will go for it

**Comment by Nickvr628** [comment]
Two points: There are thousands of possible combinations so guesswork is incredibly unlikely, and you need the nuke disk inside to start inputting codes.

This means the random clown/greytider cannot just sit at the machine and try to brute force his way in, he would need to have stolen the nuke disk first. This would result in a ban in almost any case of this occurring. If the captain wanted to try it without a good reason that would probably prompt some form of ban as well. 

Besides, not wanting to add something because someone could break the rules with it is a very weak argument.

**Comment by SomeguyManperson** [comment]
having the reason to add something being specifically so you can break the rules with it is an even weaker argument

**Comment by ynot01** [comment]
>  A copy of the code can be found in the operations room of centcom.

are you this retarded

**Comment by Nickvr628** [comment]
EDIT: I was unaware ghosts could examine papers to read them, I will remove the code paper from centcom

**Comment by OppressedCactus** [comment]
There are so many combinations that it would be super impractical to guess them all, even if they did manage to guess it (after stealing the nuke disk) I would have a hard time believing it wouldn't be stopped by an admin quickly.

**Comment by boodaliboo** [comment]
the chances are 9 x 9 x 9 x 9 x 9 or something i dunno

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 1]
```diff
@@ -42,6 +46,8 @@
 	update_icon()
 	GLOB.poi_list |= src
 	previous_level = get_security_level()
+	if(r_code == null)
```
Comment (JamieD1) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:49]: if(!r_code)

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 2]
```diff
@@ -712,3 +718,15 @@ This is here to make the tiles around the station mininuke change when it's arme
 /obj/item/disk/nuclear/fake/obvious
 	name = "cheap plastic imitation of the nuclear authentication disk"
 	desc = "How anyone could mistake this for the real thing is beyond you."
+
+
```
Comment (JamieD1) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:722]: SPACES

File: code/modules/antagonists/nukeop/nukeop.dm [hunk 3]
```diff
@@ -231,10 +228,8 @@
 		var/obj/machinery/nuclearbomb/selfdestruct/nuke = locate() in GLOB.nuke_list
 		if(nuke)
 			nuke_team.tracked_nuke = nuke
-			if(nuke.r_code == "ADMIN")
-				nuke.r_code = nuke_team.memorized_code
-			else //Already set by admins/something else?
-				nuke_team.memorized_code = nuke.r_code
+			nuke.r_code = nuke_team.memorized_code
+
```
Comment (JamieD1) [inline code/modules/antagonists/nukeop/nukeop.dm:232]: SPACE

**Comment by Alexandria404** [comment]
if you want to make QOL for admins make it a button that generates one
else you are just trying to bait people to guess it

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 4]
```diff
@@ -1,3 +1,5 @@
+var/nuke_code = random_nukecode()
```
Comment (JamieD1) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:1]: Global var bad,

Can you do this another way?

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 5]
```diff
@@ -712,3 +718,15 @@ This is here to make the tiles around the station mininuke change when it's arme
...
+
+
+//==========CENTCOM NUKE CODE BACKUP==========
+
+/obj/item/paper/fluff/stations/centcom/nuke_code_paper
+	name = "paper- 'NUCLEAR CODE: DO NOT LOSE'"
+	icon_state = "docs_blue"
+	info = null
+	infolang = null
+	New()
```
Comment (MCHSL) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:730]: replace with `/obj/item/paper/fluff/stations/centcom/nuke_code_paper/Initialize()`

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 6]
```diff
@@ -712,3 +718,15 @@ This is here to make the tiles around the station mininuke change when it's arme
...
+		..()
```
Comment (MCHSL) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:731]: replace with `. = ..()`

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm (hunk 4 above)
Comment (MCHSL) [reply code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:1]: replace with `GLOBAL_VAR_INIT(nuke_code, random_nukecode())` and add `GLOB.` before all instances of `nuke_code`

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm [hunk 7]
```diff
@@ -712,3 +718,15 @@ This is here to make the tiles around the station mininuke change when it's arme
...
 	name = "cheap plastic imitation of the nuclear authentication disk"
 	desc = "How anyone could mistake this for the real thing is beyond you."
+
+
+//==========CENTCOM NUKE CODE BACKUP==========
+
+/obj/item/paper/fluff/stations/centcom/nuke_code_paper
+	name = "paper- 'NUCLEAR CODE: DO NOT LOSE'"
+	icon_state = "docs_blue"
+	info = null
```
Comment (MCHSL) [inline code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:728]: probably can remove this and the line below

**Comment by Marmio64** [comment]
you're going to at least have to add something to the self destruct that will make it stop you for a while if you keep guessing wrong codes so it isn't abused so much.

Like, a 30 second stopper for guessing wrong 3 times in a row.

**Comment by aaqe** [comment]
Who needs this?

**Comment by alexkar598** [comment]
It would take on average 50 minutes to brute force the code at 10 tries per second ... which is alt but like... Mods cant even send erts, I'm not sure this is needed

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm (hunk 1 above)
Comment (Hopekz) [reply code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:49]: My pupil

**Comment by JamieD1** [comment]
To be honest,

All this does is let mods have the ability to give out nuke codes.

**Comment by TheGamerdk** [comment]
Mods can already give nuke codes. They just need a request to be made. Mods shouldn't be giving out codes if no request is out anyways

**Comment by Ktlwjec1** [comment]
if a maint merges this please fire them. also it literally takes like 5 seconds to set the code it's really not a hassle at all.

**Comment by Takahiru** [comment]
If i ever get hijack id gladly steal the nuke code and sit in maint for an hour to brute force it, why would you add this?

**Comment by deathslayer2481** [comment]
I will say that ‘random’ is never truelly ‘random’.

**Comment by Hopekz** [comment]
This is an admin quality of life PR.
The reception this PR is getting both here and on discord from the admins that this PR is supposed to be benefiting is very negative.
Closing this PR.

File: code/modules/antagonists/nukeop/equipment/nuclearbomb.dm (hunk 1 above)
Comment (Amelia0010) [reply code/modules/antagonists/nukeop/equipment/nuclearbomb.dm:49]: image

**Pull request closed without merging**